File: 프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.java
import java.util.Arrays;

    class Solution {
public static int solution(int[] wallet, int[] bill) {
int answer = 0;
Arrays.sort(wallet);
Arrays.sort(bill);

while (!checkSize(wallet, bill)) {
fold(wallet, bill);
answer++;
}
return answer;
}

private static void fold(int[] wallet, int[] bill) {
if (bill[0] > bill[1]) {
bill[0] /= 2;
} else {
bill[1] /= 2;
} 
}
private static boolean checkSize(int[] wallet, int[] bill) {
Arrays.sort(wallet);
Arrays.sort(bill);

return bill[0] <= wallet[0] && bill[1] <= wallet[1];
}
    }

File: 프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.py
def solution(wallet, bill):
    answer = 0
    
    # 종료조건 어느 한쪽이 지갑보다 큰 경우에는 무한 반복
    while min(bill) > min(wallet) or max(bill) > max(wallet):
        
        # 더 큰쪽 찾아서 접기
        if bill[0] > bill[1]:
            bill[0] = bill[0] // 2
        else:
            bill[1] = bill[1] // 2
        
        # 정답 +1
        answer += 1
        
    return answer